CycloneDX/Program.cs

```diff
@@ -257,5 +284,48 @@ public static async Task<int> Main(string[] args)
 
             return 0;
         }
+
+        static public Bom ReadMetaDataFromFile(Bom bom, string templatePath)
+        {
+            try
+            {
+                return XmlBomDeserializer.Deserialize(File.ReadAllText(templatePath));
+            }
+            catch (IOException ex)
+            {
+                Console.Error.WriteLine($"Could not read Metadata file.");
+                Console.WriteLine(ex.Message);
+            }
+            return bom;
+        }
+        static public void AddMetadataTool(Bom bom)
+        {
+            string toolname = "CycloneDX module for .NET";
```

Comment:
Why does this default to "CycloneDX module for .NET"? The tools cannot simply make thing up.

Reply:
I used the title from [README.md](https://github.com/CycloneDX/cyclonedx-dotnet/blob/master/README.md#cyclonedx-module-for-net) and this describes the tool which was used to create the BOM. The [CycloneDX Node.js Module](https://www.npmjs.com/package/@cyclonedx/bom) creates a similar **Tool** entry. 
My preferred choice would be the details from **CycloneDX.exe** which uses product name "_CycloneDX_" but this is not helpful.

![image](https://user-images.githubusercontent.com/2315215/106719609-cec64480-6602-11eb-94f0-e5dd329dbe3b.png)

Reply:
@stevespringett Do you have any proposals or recommendation for this PR?

Reply:
The metadata tools shows the creator tool for the BOM (This is similar to npm tool)